client/handlers.go

```diff
@@ -122,9 +122,19 @@ func (v2 *signer) Sign() error {
 		return err
 	}
 	host, canonicalPath := parsedURL.Host, parsedURL.Path
+	// Host may not be parsed successfully by method.
+	if len(host) < 1 {
+		host = v2.Request.URL.Host
+	}
 	v2.Request.Header["Host"] = []string{host}
 	v2.Request.Header["x-amz-date"] = []string{v2.Time.In(time.UTC).Format(time.RFC1123)}
 
+	// Alibaba Cloud OSS date's formate must be http.TimeFormat
+	if strings.Contains(host, "aliyuncs.com") {
```

Comment:
Not sure what the solution is here, but feels like we're duplicating logic between this and the `config.go` code with regards to knowing what an AliCloud host is. We do something similar in `isAlicloudHost` over there, can we somehow share that or move this logic into config?